contracts/0.4.24/StETH.sol

```diff
@@ -291,21 +321,46 @@ contract StETH is IERC20, Pausable {
         }
     }
 
+    /**
+     * @notice Moves `_sharesAmount` token shares from the caller's account to the `_recipient` account.
+     *
+     * @return amount of transferred tokens.
+     * Emits a `TransferShares` event.
+     * Emits a `Transfer` event.
+     *
+     * Requirements:
+     *
+     * - `_recipient` cannot be the zero address.
+     * - the caller must have at least `_sharesAmount` shares.
+     * - the contract must not be paused.
+     *
+     * @dev The `_sharesAmount` argument is the amount of shares, not tokens.
+     */
+    function transferShares(address _recipient, uint256 _sharesAmount) public returns (uint256) {
+        _transferShares(msg.sender, _recipient, _sharesAmount);
+        emit TransferShares(msg.sender, _recipient, _sharesAmount);
```

Comment:
Maybe it's better to put the emitting of the `TransferShares` event in the `_transferShares()` method? It might help reduce the bytecode of the contract and remove redundant code.

Reply:
+1 This change forces a number of other changes.. Preparing a separate PR to address this.

Reply:
Here is a rough sketch which mostly aims to highlight the scope of the changes required https://github.com/lidofinance/lido-dao/pull/421

Reply:
I suggest leaving this improvement for a later upgrade since we need to finally freeze the code for audit

Reply:
Created issue #422

Reply:
It's provisionally scheduled for the next protocol upgrade :+1: